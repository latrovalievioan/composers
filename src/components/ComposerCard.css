.composer-card {
    display: flex;
    position: relative;
    background: var(--orange);
    border-radius: 1rem;
    overflow: hidden;
    max-height: 200px;
}

.composer-card > img {
    object-fit: fill;
}


.composer-info {
    display: flex;
    flex-direction: column;
    color: var(--light);
    padding: 1rem;
}

.composer-title {
    margin-bottom: auto;
    margin-top: 0.5rem;
}

.image-overlay {
    position: absolute;
    display: flex;
    flex-direction: column;
    width: 200px;
    height: 200px;
    align-items: center;
    justify-content: center;
    color: var(--light);
    background: #0000007a;
    opacity: 0;
    transition: opacity 0.5s;
}

.image-overlay:hover {
    opacity: 1;
}

.plus-icon {
    cursor:pointer;
    margin-bottom: 0.4rem;
}

.plus-icon:hover {
    color: var(--orange);
}

p {
    cursor: context-menu;
}