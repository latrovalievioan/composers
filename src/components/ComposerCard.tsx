import React, { ReactElement } from 'react';
import { Composer } from '../types';
import './ComposerCard.css';
import { AddRemoveButton } from './AddRemoveButton';

export const ComposerCard = ({ composer }: { composer: Composer }) => {
  return (
    <div className="composer-card">
      <img src={composer.portrait} alt=""></img>
      <div className="composer-info">
        <h3 className="composer-title">{composer.complete_name}</h3>
        <span>Birth: {composer.birth}</span>
        <span>Death: {composer.death || 'still alive'}</span>
        <span>Epoch: {composer.epoch}</span>
        <AddRemoveButton composer={composer} />
      </div>
    </div>
  );
};
