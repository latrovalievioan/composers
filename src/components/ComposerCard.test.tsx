import React from 'react';
import renderer from 'react-test-renderer';
import { ComposerCard } from './ComposerCard';

jest.mock('./AddRemoveButton', () => {
  return { AddRemoveButton: () => <div id="mock-button"></div> };
});

it('renders correctly', () => {
  const nav = renderer
    .create(
      <ComposerCard
        composer={{
          id: '123123',
          name: 'tsivanoff',
          complete_name: 'ts!van-0ff;)',
          birth: 'yesterday',
          death: 'tomorrow',
          epoch: 'paleolit',
          portrait: 'https://i.ytimg.com/vi/cNycdfFEgBc/maxresdefault.jpg',
        }}
      />
    )
    .toJSON();
  expect(nav).toMatchSnapshot();
});
