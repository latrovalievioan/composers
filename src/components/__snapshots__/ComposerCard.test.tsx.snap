// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="composer-card"
>
  <img
    alt=""
    src="https://i.ytimg.com/vi/cNycdfFEgBc/maxresdefault.jpg"
  />
  <div
    className="composer-info"
  >
    <h3
      className="composer-title"
    >
      ts!van-0ff;)
    </h3>
    <span>
      Birth: 
      yesterday
    </span>
    <span>
      Death: 
      tomorrow
    </span>
    <span>
      Epoch: 
      paleolit
    </span>
    <div
      id="mock-button"
    />
  </div>
</div>
`;
